modules/diff.js

```diff
@@ -73,27 +74,36 @@ const getGitDiff = async ({ product, base, head = 'master', dev, prod }) => {
   )
 
   const contributors = new Set([])
-  const breaking = new Set([])
-  const noMerges = commits.filter(({ parents }) => parents.length <= 1).map(({
+  const noMerges = commits
+    .filter(({ parents }) => parents.length <= 1)
+    .reverse()
+    .map(({ sha, html_url, commit }) => ({ sha, html_url, commit }))
+  const logs = noMerges.map(({
     sha,
-    html_url,
     commit: { message, author, committer },
+    html_url,
   }) => {
-    const { name = '`Unknown`' } = (author || committer || {})
-    contributors.add(name) // add to contributors set
-    // highlight commits that may indicate breaking changes
-    let msg = message
-    if (mayBreak(message)) {
-      msg = `*${message}*`
-      breaking.add(sha)
-    }
-    return {
-      type: 'section',
-      text: { type: 'mrkdwn', text: `${msg} (<${html_url}|${sha.slice(0, 7)}> by ${name})` },
-    }
+    // %h::%s::%b::%an||
+    const [s, ...b] = message.split('\n')
+    const { name: an = '`Unknown`' } = (author || committer || {})
+    contributors.add(an) // side-effect to populate contributors
+    return [`<${html_url}|${sha.slice(0, 7)}>`, s, b.join('\n'), an].join('::')
+  })
+  const parsed = (await parseCommits({ logs })).reduce((acc, { labels: [label], t1, msg: { s, t2 } }) => {
+    acc[label] = acc[label] || []
+    acc[label].push(`${[t1, t2].filter(t => t).join('/')} - ${s}`)
+    return acc
+  }, {})
+  // TODO: need to extend release formatter to support slack blocks mrkdwn format too
+  // const formatted = formatChangelog({ parsed, version: head.slice(0, 7), previous: base.slice(0, 7) })
+  let formatted = `*Changelog: from ${base.slice(0, 7)} to ${head.slice(0, 7)}*\n`
+  Object.entries(parsed).forEach(([label, items]) => {
+    formatted += `\n*${label}*\n`
+    items.forEach((item) => {
+      formatted += `* ${item}\n`
```

Comment:
lol don't forget that `*` is [not parsing](https://github.com/EQWorks/legion/blob/61cabb11f6a64636106000cc9750aff79c742191/modules/notes.js#L194) into list items as we talked about once - slack limitation

Reply:
it's not but i think sometimes it's _okay_ to see the raw form